example/offboard_velocity/offboard_velocity.cpp

```diff
@@ -0,0 +1,229 @@
+//
+// Example that demonstrates Offboard velocity control using LOCAL_NED & BODY_NED
+//
+// Author: Julian Oes <julian@oes.ch>, Shakthi Prashanth <shakthi.prashanth.m@intel.com>
+#include <iostream>
+#include <cmath>
+#include <thread>
+#include <chrono>
+#include <dronecore/dronecore.h>
+
+using namespace dronecore;
+
+#define ERROR_CONSOLE_TEXT "\033[31m" //Turn text on console red
+#define TELEMETRY_CONSOLE_TEXT "\033[34m" //Turn text on console blue
+#define NORMAL_CONSOLE_TEXT "\033[0m"  //Restore normal console colour
+
+void wait_msec(int msec)
+{
+    std::this_thread::sleep_for(std::chrono::milliseconds(msec));
+}
+
+void wait_sec(int sec)
+{
+    std::this_thread::sleep_for(std::chrono::seconds(sec));
+}
+
+int main(int argc, char **argv)
+{
+    DroneCore dc;
+    std::string offb_mode = (argc > 1) ? argv[1] : "";
+
+    if (argc == 1 || (offb_mode != "local-ned" && offb_mode != "body-ned")) {
+        std::cerr << "Usage: offboard <coordinate-frame>\n" <<
+                  "coordinate-frame: \"local-ned\", or \"body-ned\""
+                  << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    // add udp connection
+    DroneCore::ConnectionResult ret = dc.add_udp_connection();
+    if (ret != DroneCore::ConnectionResult::SUCCESS) {
+        std::cout << ERROR_CONSOLE_TEXT << "Connection failed: "
+                  << DroneCore::connection_result_str(ret)
+                  << NORMAL_CONSOLE_TEXT << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    // Wait for device to connect via heartbeat.
+    wait_sec(2);
+    Device &device = dc.device();
+
+    while (!device.telemetry().health_all_ok()) {
+        std::cout << "waiting for device to be ready" << std::endl;
+        wait_sec(1);
+    }
+
+    Action::Result arm_result = device.action().arm();
+    if (arm_result != Action::Result::SUCCESS) {
+        std::cout << ERROR_CONSOLE_TEXT << "Arming failed:" << Action::result_str(
+                      arm_result) << NORMAL_CONSOLE_TEXT << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    Action::Result takeoff_result = device.action().takeoff();
+    if (takeoff_result != Action::Result::SUCCESS) {
+        std::cout << ERROR_CONSOLE_TEXT << "Takeoff failed:" << Action::result_str(
+                      takeoff_result) << NORMAL_CONSOLE_TEXT << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    std::cout << "In Air" << std::endl;
+    wait_sec(5);
```

Comment:
you could use `[in_air_async](https://github.com/dronecore/DroneCore/blob/develop/plugins/telemetry/telemetry_impl.h#L65)` instead of waiting

Reply:
Generally I don't see much use for the async APIs for managing a single vehicle. 

In any case though we might use this as an opportunity to show off some "comprehensive" takeoff code, as per [this PR](https://github.com/dronecore/docs/pull/42)
- Close app if vehicle not calibrated
- Print what we're waiting on (maybe)
- Check that arming occurred before calling takeoff
- Check that we reached takeoff height rather than using timer. This is a bit better than in_air() because you actually know you have completed the takeoff.

Reply:
@mrpollo, this is basically to let drone reach takeoff altitude before we start issuing OFFBOARD commands. I think it should be fine to sleep for 5 seconds, similar to the way done in [takeoff land](https://github.com/dronecore/DroneCore/blob/master/example/takeoff_land/takeoff_and_land.cpp#L96) example.